ServerlessLibraryUI/src/components/ConsentModal.vue

```diff
@@ -57,26 +57,45 @@
 
   methods: {
     close() {
+      this.trackEvent('/sample/deploy/cancel');
+      this.closeModal();
+    },
+
+    closeModal() {
       this.$emit('close');
     },
 
-    getDeployUrl(item) {
-      return 'https://portal.azure.com/#create/Microsoft.Template/uri/' + encodeURIComponent(item.template);
+    getDeployUrl() {
+      return 'https://portal.azure.com/#create/Microsoft.Template/uri/' + encodeURIComponent(this.data.template);
     },
 
-    outboundRepoClick(data) {
+    outboundDeployClick() {
+      this.trackEvent('/sample/deploy/agree');
       fetch('https://www.serverlesslibrary.net/api/Library'
       , {
           method: 'PUT',
-          body:'"' + data.template + '"',
+          body:'"' + this.data.template + '"',
           headers: {
           "Content-Type": "application/json"
           },
         })
         .then(response => response.body)
         .catch((err)=>console.error(err.message));
-        this.close();
+        this.closeModal();
     },
+
+    trackEvent(eventName) {
+      if (typeof appInsights !== 'undefined') {
+        const eventData = this.getDataToLog();
+        appInsights.trackEvent(eventName, eventData);
+      }
+    },
+
+    getDataToLog() {
```

Comment:
why a single line method?

Reply:
Mainly to keep it consistent with the other Vue components. Also, in future we might want to log additional data with this event.